context/BookContext.tsx

```diff
@@ -8,38 +8,92 @@ import React, {
 } from "react"
 
 import { BookType } from "api/types"
+import { ActivePageEnum } from "api/enums"
 
 type BookContextType = {
   discoverBooks: BookType[]
   finishedBooks: BookType[]
   readingBooks: BookType[]
+  selectedBook: BookType
+  isBookSelected: boolean
+  active: ActivePageEnum | string
   setDiscoverBooks: Dispatch<SetStateAction<BookType[]>>
   setFinishedBooks: Dispatch<SetStateAction<BookType[]>>
   setReadingBooks: Dispatch<SetStateAction<BookType[]>>
+  setSelectedBook: Dispatch<SetStateAction<BookType>>
+  setIsBookSelected: Dispatch<SetStateAction<boolean>>
+  setActive: Dispatch<SetStateAction<ActivePageEnum | string>>
+  onRemove: (book: BookType, list_type: string) => void
 }
 
 const BookContext = createContext<BookContextType | null>(null)
 
 const useBooks = () => {
   const context = useContext(BookContext)
   if (!context) {
-    throw new Error(`useAlertModals must be used within an AlertModalContext`)
+    throw new Error(`useBooks must be used within a BookContextProvider`)
```

Comment:
use your error handling shared function instead.

Reply:
![image](https://user-images.githubusercontent.com/91857581/191431985-2e276643-1779-4563-8380-a0cfd2eff3d1.png)

getting this error inside components where `useBooks` is declared. This happens when I try to swap `throw new Error` with the shared `errorHandler` there